opm/simulators/wells/VFPHelpers.cpp

```diff
@@ -0,0 +1,656 @@
+/*
+  Copyright 2015 SINTEF ICT, Applied Mathematics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+#include <opm/simulators/wells/VFPHelpers.hpp>
+
+#include <opm/common/ErrorMacros.hpp>
+
+#include <opm/material/densead/Evaluation.hpp>
+
+#include <opm/parser/eclipse/EclipseState/Schedule/VFPInjTable.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/VFPProdTable.hpp>
+
+#include <cassert>
+#include <cmath>
+#include <stdexcept>
+
+namespace {
+
+/**
+ * Helper function that finds x for a given value of y for a line
+ * *NOTE ORDER OF ARGUMENTS*
+ */
+double findX(const double& x0,
+             const double& x1,
+             const double& y0,
+             const double& y1,
+             const double& y)
```

Comment:
Any reason why the input parameters are reference to `double` instead of plain `double`?

Reply:
i wouldn't know, but i can't imagine there is a reason (i just shuffled the code around).

Reply:
There are multiple places used `const double&`, since you are here, maybe it is good to stop them from circulating.

Reply:
done.